pkg/resources/execute_acceptance_test.go

```diff
@@ -740,3 +791,59 @@ func verifyGrantExists(t *testing.T, roleId sdk.AccountObjectIdentifier, privile
 		return nil
 	}
 }
+
+func TestAcc_Execute_ImportWithRandomId(t *testing.T) {
+	id := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	newId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					_, databaseCleanup := acc.TestClient().Database.CreateDatabaseWithIdentifier(t, id)
+					t.Cleanup(databaseCleanup)
+				},
+				Config:                  executeConfigCreateDatabase(id),
+				ResourceName:            "snowflake_execute.test",
+				ImportState:             true,
+				ImportStatePersist:      true,
+				ImportStateId:           "random_id",
+				ImportStateVerifyIgnore: []string{"query_results"},
+			},
+			// filling the empty state fields (execute changed from empty)
+			{
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_execute.test", plancheck.ResourceActionUpdate),
+					},
+				},
+				Config: executeConfigCreateDatabase(id),
```

Comment:
will it be run in this step or not? Asking because not everyone uses IF NOT EXISTS, so I would like to make sure it will work

Reply:
The state was persisted from the previous state. It doesn't run execute because if it would the 2nd step would produce an error that the database already exists and cannot be created.